backend/services/stats_service.py

```diff
@@ -479,3 +482,114 @@ def update_project_stats(project_id: int):
             Task.task_status == TaskStatus.BADIMAGERY.value
         ).count()
         project.save()
+
+    @staticmethod
+    def get_task_stats(
+        start_date, end_date, org_id, org_name, campaign_id, project_id, country
+    ):
+        """ Creates tasks stats for a period using the TaskStatsDTO """
+
+        query = (
+            db.session.query(
+                TaskHistory.task_id,
+                TaskHistory.project_id,
+                TaskHistory.action_text,
+                func.DATE(TaskHistory.action_date).label("day"),
+            )
+            .filter(
+                TaskHistory.action == "STATE_CHANGE",
+                or_(
+                    TaskHistory.action_text == "MAPPED",
+                    TaskHistory.action_text == "VALIDATED",
+                    TaskHistory.action_text == "BAD_IMAGERY",
+                ),
+            )
+            .filter(
+                and_(
+                    func.DATE(TaskHistory.action_date) >= start_date,
+                    func.DATE(TaskHistory.action_date) <= end_date,
+                )
+            )
+            .group_by(
+                TaskHistory.action_text,
+                "day",
+                TaskHistory.project_id,
+                TaskHistory.task_id,
+            )
+            .order_by("day")
+        )
+
+        if org_id:
+            query = query.join(Project, Project.id == TaskHistory.project_id).filter(
+                Project.organisation_id == org_id
+            )
+        if org_name:
+            organisation = OrganisationService.get_organisation_by_name(org_name)
+            query = query.join(Project, Project.id == TaskHistory.project_id).filter(
+                Project.organisation_id == organisation.id
```

Comment:
We need to handle better the case when there is not an organisation with that name. I suggest:

```
            try:
                organisation_id = OrganisationService.get_organisation_by_name(org_name).id
            except NotFound:
                organisation_id = None
            query = query.join(Project, Project.id == TaskHistory.project_id).filter(
                Project.organisation_id == organisation_id
            )
```